src/LinearRemapFV.cpp

```diff
@@ -2662,164 +2825,584 @@ void LinearRemapIntegratedBilinear(
+			}
+	
+			// This Face
+			const Face & faceFirst = meshInput.faces[ixFirst];
+	
+			// Find the set of Faces that overlap faceFirst
+			int ixOverlapBegin = ixOverlap;
+			int ixOverlapEnd = ixOverlapBegin;
+	
+			for (; ixOverlapEnd < meshOverlap.faces.size(); ixOverlapEnd++) {
+				if (meshOverlap.vecSourceFaceIx[ixOverlapEnd] != ixFirst) {
+					break;
+				}
+			}
+			
+			int nOverlapFaces = ixOverlapEnd - ixOverlapBegin;
+	
+			if( nOverlapFaces == 0 ){ 
+				
+				continue;
+				
+			}
+	
+			// Loop through all overlap faces associated with this source face
+			for (int j = 0; j < nOverlapFaces; j++) {
+				
+				int iTargetFace = meshOverlap.vecTargetFaceIx[ixOverlap + j];
+	
+				// signal to not participate, because it is a ghost target
+				if( iTargetFace < 0 ) continue;  // skip and do not do anything
+	
+				const Face & faceOverlap = meshOverlap.faces[ixOverlap + j];
+	
+				int nSubTriangles = faceOverlap.edges.size() - 2;
+	
+				// Jacobian at each quadrature point
+				DataArray2D<double> dQuadPtWeight(nSubTriangles, dW.GetRows());
+				DataArray2D<Node> dQuadPtNodes(nSubTriangles, dW.GetRows());
+	
+				// Compute quadrature area of each overlap face
+				double dQuadratureArea = 0.0;
+	
+				for (int k = 0; k < nSubTriangles; k++) {
+					for (int p = 0; p < dW.GetRows(); p++) {
+	
+						dQuadPtWeight(k,p) =
+							CalculateSphericalTriangleJacobianBarycentric(
+								meshOverlap.nodes[faceOverlap[0]],
+								meshOverlap.nodes[faceOverlap[k+1]],
+								meshOverlap.nodes[faceOverlap[k+2]],
+								dG(p,0), dG(p,1),
+								&(dQuadPtNodes(k,p)));
+	
+						dQuadPtWeight(k,p) *= dW[p];
+	
+						dQuadratureArea += dQuadPtWeight(k,p);
+					}
+				}
+	
+				// Loop through all sub-triangles of this overlap Face
+				for (int k = 0; k < nSubTriangles; k++) {
+	
+					// Loop through all quadrature nodes on this overlap Face
+					for (int p = 0; p < dW.GetRows(); p++) {
+	
+						// Get quadrature node and pointwise Jacobian
+						Node & nodeQ = dQuadPtNodes(k,p);
+	
+						//Get the dual mesh face whose center is nearest to the sample point
+	
+						kdres * kdresTarget =
+							kd_nearest3(
+								kdTarget,
+								nodeQ.x,
+								nodeQ.y,
+								nodeQ.z);
+								
+						
+						Face * pFace = (Face *)(kd_res_item_data(kdresTarget));
+		
+						int iNearestFace = pFace - &(meshInputDual.faces[0]);
+						
+						// Find face that contains nodeQ
+						// This is the face whose local index is iFaceFinal
+						
+						int iFaceFinal = 0;
+						
+						GetFaceThatContainsPoint(meshInputDual,iNearestFace,iFaceFinal,nodeQ.x,nodeQ.y,nodeQ.z);
+						
+						int iEdges = meshInputDual.faces[iFaceFinal].edges.size();
+						
+						//NodeVector nodesCurrentFace(iEdges);
+						NodeVector nodesCurrentFace;
+						
+						for (int i = 0; i < iEdges; i++){
+							
+							nodesCurrentFace.push_back(meshInputDual.nodes[meshInputDual.faces[iFaceFinal][i]]);
```

Comment:
The diffs are absolutely terrible in the interface. The local git diff isn't any better unfortunately, which makes it very hard to track new code changes in the `LinearRemapIntegratedBilinear` routine. As far as my review goes, I don't see anything significantly out of the ordinary in the new additions, but if there is a specific chunk of code I should review, let me know.

Reply:
Most of the changes are in the functions LinearRemapIntegratedBilinear and LinearRemapIntegratedGeneralizedBarycentric.  Originally, these functions only worked for global source meshes, so they had to be altered significantly to account for source meshes with holes.  Many of the other changes involved simply replacing blocks of code with calls to functions.  For instance, the bilinear and barycentric weights are now obtained by calling the functions “BilinearWeights” and “GeneralizedBarycentricCoordinates,” respectively.  

For the most part the LinearRemapIntegratedBilinear and LinearRemapIntegratedGeneralizedBarycentric functions are similar to their non-integrated counterparts in LinearRemapBilinear and LinearRemapGeneralizedBarycentric, apart from the fact that the integrated versions loop over the overlap mesh, instead of the target mesh.

Reply:
Thanks for the clarifications. I reviewed the code again and the changes are similar to the non-integrated variants as you mentioned.